### PR TITLE
scheduler optimization - worker precompile verification

### PR DESCRIPTION
**Pull request by apfitzge**
#### Problem
scheduler is single threaded. doing precompile verification during the intial sanitization is thus also single-threaded.

#### Summary of Changes
move precompile verification to worker threads to improve efficiency

Fixes #

### CODE REVIEW

**Review by jstarry — APPROVED** [review]

File: core/src/banking_stage/consumer.rs [hunk 1]
```diff
@@ -413,12 +413,20 @@ impl Consumer {
...
-        let check_results = check_results.into_iter().map(|result| result.map(|_| ()));
+        // If checks passed, verify pre-compiles and continue processing on success.
+        let check_results: Vec<_> = txs
+            .iter()
+            .zip(check_results)
+            .map(|(tx, result)| match result {
+                Ok(_) => tx.verify_precompiles(&bank.feature_set),
+                Err(err) => Err(err),
+            })
+            .collect();
```
Comment (sakridge) [inline core/src/banking_stage/consumer.rs:424]: Does this need to add another allocation?

File: core/src/banking_stage/immutable_deserialized_packet.rs [hunk 2]
```diff
@@ -103,10 +103,10 @@ impl ImmutableDeserializedPacket {
     }
 
     // This function deserializes packets into transactions, computes the blake3 hash of transaction
-    // messages, and verifies secp256k1 instructions.
+    // messages.
     pub fn build_sanitized_transaction(
         &self,
-        feature_set: &Arc<feature_set::FeatureSet>,
+        _feature_set: &Arc<feature_set::FeatureSet>,
```
Comment (sakridge) [inline core/src/banking_stage/immutable_deserialized_packet.rs:109]: why not remove? reduce backport diff?

Comment (jstarry) [reply core/src/banking_stage/immutable_deserialized_packet.rs:109]: Yup exactly

File: core/src/banking_stage/consumer.rs (hunk 1 above)
Comment (jstarry) [reply core/src/banking_stage/consumer.rs:424]: Discussed offline that this is to ensure that downstream timing isn't affected by this extra verification

Comment (sakridge) [reply core/src/banking_stage/consumer.rs:424]: ok, probably deserves a comment at least

**Pull request merged**